packages/beacon-node/src/node/nodejs.ts

```diff
@@ -187,6 +185,31 @@ export class BeaconNode {
       signal.addEventListener("abort", metrics.close, {once: true});
     }
 
+    const clock = new Clock({config, genesisTime: anchorState.genesisTime, signal});
+
+    // Prune hot db repos
+    // TODO: Should this call be awaited?
+    await db.pruneHotDb();
+
+    if (opts.chain.pruneHistory) {
+      // prune ALL stale data before starting
+      await callFnWhenAwait(
+        pruneHistory(
+          config,
+          db,
+          logger,
+          metrics,
+          anchorState.finalizedCheckpoint.epoch,
+          clock.currentEpoch,
+          Infinity,
+          Infinity
+        ),
+        () => logger.info("Pruning historical data..."),
```

Comment:
we might still want the initial log so we know pruning is enabled, this will only be logged if it takes >30 seconds